read-fonts/src/collections/int_set/mod.rs

```diff
@@ -2536,4 +2624,47 @@ mod test {
 
         assert_eq!(a.cmp(&b), Ordering::Less);
     }
+
+    fn roundtrip_json<T: Domain>(set: &IntSet<T>) -> Result<IntSet<T>, serde_json::Error> {
+        let json = serde_json::to_vec(&set).unwrap();
+        serde_json::from_slice(&json)
+    }
+
+    #[test]
+    fn simple_serde() {
+        let mut set = IntSet::empty();
+        set.insert(0u32);
+        set.insert(u32::MAX);
+        assert_eq!(roundtrip_json(&set).unwrap(), set);
+    }
+
+    #[test]
+    fn serde_inverted() {
+        let mut set = IntSet::all();
+        set.remove_range(0u16..=420);
+        let bytes = serde_json::to_string(&set).unwrap();
+        assert!(bytes.len() < 5000, "sanity check serialization");
+        assert_eq!(roundtrip_json(&set).unwrap(), set)
+    }
+
+    #[test]
+    fn serde_inverted_out_of_domain() {
```

Comment:
Also add an out of domain test on:
- a non-continous domain (like EvenInts) 
- Inverted non-continous domain
- Inverted continous domain (should fail during deserialize with the changes I suggested above)